src/ctap/mod.rs

```diff
@@ -1017,6 +1031,23 @@ where
     }
 
     fn process_get_info(&self) -> Result<ResponseData, Ctap2StatusCode> {
+        let has_always_uv = self.persistent_store.has_always_uv()?;
+        #[cfg(feature = "with_ctap1")]
+        let mut versions = vec![
+            String::from(FIDO2_VERSION_STRING),
+            String::from(FIDO2_1_VERSION_STRING),
+        ];
+        #[cfg(feature = "with_ctap1")]
+        {
+            if !has_always_uv {
+                versions.insert(0, String::from(U2F_VERSION_STRING))
+            }
+        }
+        #[cfg(not(feature = "with_ctap1"))]
+        let versions = vec![
+            String::from(FIDO2_VERSION_STRING),
+            String::from(FIDO2_1_VERSION_STRING),
+        ];
```

Comment:
I guess this is convoluted because of the "mut" that should only be present with ctap1. What about disabling the lint for that line instead and only define the versions variable once? The lint may actually be disable only when not ctap1.

Reply:
Done!